hailjwt/hailjwt/hailjwt.py

```diff
@@ -97,3 +77,23 @@ def wrapped(request, userdata, *args, **kwargs):
             return fun(request, *args, **kwargs)
         raise web.HTTPNotFound()
     return wrapped
+
+
+def authenticated_users_only_headers(fun):
+    token_getter = lambda request: request.headers.get('user')
```

Comment:
You should use the `Authorization` header

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization

with the Bearer authorization scheme, which means the header looks like:

`Authorization: Bearer <token>`

https://jwt.io/introduction/, and for the technical details: https://tools.ietf.org/html/rfc6750

I have a slight prefer for semantic rather than operational names.  The headers are support to protect REST API endpoints, the cookies to protect web endpoints.  So maybe `rest_authenticated_users_only` and `web_authenticated_users_only`?

Reply:
Thanks for sharing these. I learned a lot by reading them!